api/errors/builder.go

```diff
@@ -0,0 +1,93 @@
+package errors
+
+// This file contains all of the errors that can be generated from the
+// docker/builder component.
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution/registry/api/errcode"
+)
+
+var (
+	// ErrorCodeAtLeastOneArg is generated when the parser comes across a
+	// Dockerfile command that doesn't have any args.
+	ErrorCodeAtLeastOneArg = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
This seems like a mis-use of the `errcode` package. Are there different actions a client might take under each case? Seems like there are only a few real conditions:

1. `ArgumentError`
2. `SyntaxError`
3. `Unsupported`